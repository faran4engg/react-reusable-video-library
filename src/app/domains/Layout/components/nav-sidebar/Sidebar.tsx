import { FC } from 'react';
import { SidebarProps } from './types';
import SidebarContent from './SidebarContent';

const cls = {
  aside: isSidebarOpen =>
    `lg:flex block lg:relative fixed inset-y-0 left-0 z-30 w-44 overflow-y-auto transition duration-300 ease-out transform translate-x-0 lg:translate-x-0 lg:inset-0 shadow-right shadow-md bg-light-mode dark:bg-dark-mode ${
      isSidebarOpen ? 'ease-out translate-x-0' : 'ease-in -translate-x-full'
    } `,
};

const Sidebar: FC<SidebarProps> = ({ isSidebarOpen, handleIsSidebarOpen }) => (
  <aside className={cls.aside(isSidebarOpen)}>
    <SidebarContent handleIsSidebarOpen={handleIsSidebarOpen} />
  </aside>
);

export default Sidebar;
