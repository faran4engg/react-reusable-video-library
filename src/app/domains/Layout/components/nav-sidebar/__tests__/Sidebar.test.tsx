import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import Sidebar from '../Sidebar';

const renderer = createRenderer();

describe('<Sidebar />', () => {
  it('should render and match the snapshot when sidebar is open', () => {
    renderer.render(<Sidebar isSidebarOpen />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should render and match the snapshot when sidebar is closed', () => {
    renderer.render(<Sidebar isSidebarOpen={false} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
