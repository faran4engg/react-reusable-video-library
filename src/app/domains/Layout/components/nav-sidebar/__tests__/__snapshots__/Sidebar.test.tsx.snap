// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render and match the snapshot when sidebar is closed 1`] = `
<aside
  className="lg:flex block lg:relative fixed inset-y-0 left-0 z-30 w-44 overflow-y-auto transition duration-300 ease-out transform translate-x-0 lg:translate-x-0 lg:inset-0 shadow-right shadow-md bg-light-mode dark:bg-dark-mode ease-in -translate-x-full "
>
  <SidebarContent />
</aside>
`;

exports[`<Sidebar /> should render and match the snapshot when sidebar is open 1`] = `
<aside
  className="lg:flex block lg:relative fixed inset-y-0 left-0 z-30 w-44 overflow-y-auto transition duration-300 ease-out transform translate-x-0 lg:translate-x-0 lg:inset-0 shadow-right shadow-md bg-light-mode dark:bg-dark-mode ease-out translate-x-0 "
>
  <SidebarContent />
</aside>
`;
